KadaneAlgorithm.cpp

```diff
@@ -0,0 +1,25 @@
+#include<iostream>
+#include<vector>
+#include<algorithm>
+
+int main() {
+  using std::vector;
+  int n;
+  int currentsum = 0, maxsum = 0;  // Stores current sum and maximum sum
+  std::cin >> n;
+  vector <int> a;
+  for ( int i = 0; i < n ; ++i ) {
+      int x;
+      std::cin >> x;
+      a.push_back(x);
+  }
+  for ( int i = 0; i < n ; ++i ) {
+     currentsum = currentsum + a[i];
+     //  If current sum becomes negative make it zero
+     if ( currentsum < 0 )
+      currentsum = 0;
+     maxsum = std::max(currentsum , maxsum);
+  }
+  std::cout<< maxsum << endl;
```

Comment:
this should be `std::cout << maxsum << std::endl;` or `std::cout << maxsum << " \n ";`
better to use `"\n"` insted of `std::endl` because `std::endl` is slow

Reply:
How slow?!?

Reply:
`std::endl` is actually slower because it forces a flush, which actually unnecessary. You would need to force a flush right before prompting the user for input from cin, but not when writing a million lines of output , it's good to use `"\n"` instead of `std::endl`

Reply:
and i belive this algorithm repository is used by mostly competative programmers so it's better to maintain the performace that's why i suggested that